libsolidity/ast/Types.cpp

```diff
@@ -1281,9 +1281,9 @@ shared_ptr<FixedBytesType> FixedBytesType::smallestTypeForLiteral(string const&
 	return shared_ptr<FixedBytesType>();
 }
 
-FixedBytesType::FixedBytesType(int _bytes): m_bytes(_bytes)
+FixedBytesType::FixedBytesType(unsigned int _bytes): m_bytes(_bytes)
 {
-	solAssert(m_bytes >= 0 && m_bytes <= 32,
```

Comment:
Please remove the `>=0` check instead of changing semantics.

Reply:
Shouldn't there be an assert for `m_bytes != 0` though? In what case can it be 0?

Reply:
There was a case in the past where zero length bytes was possible, even if not as a type itself.

Reply:
The parser enforces `0 < m <= 32`, probably good idea to leave the assert as is then.

Reply:
I think `""` can have a type of `bytes0` (of course only when using `var`).

My point is: This is about a refactoring, it should not change semantics.

Reply:
Lets strengthen that in a separate PR then, but it doesn't seem to be the case:
```
var a = "";
bytes1 b = a;
```

```
Error: Type string memory is not implicitly convertible to expected type bytes1.
```